### PR TITLE
fix: WDC color APCA checks

### LINKED ISSUE #23218: Fix APCA contrast usage

### PR DESCRIPTION
**Pull request by ichik**
## Description

tl;dr Fixed APCA contrast checks for both light and dark modes (detailed info in the ticket). Refined how we pick shade or tint for `fgOnAccent`.

Fixes #23218    

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- Chore (housekeeping or task changes that don't impact user perception)

## How Has This Been Tested?

- Manual

### Test Plan
Initial POC refinement, no testing necessary

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/packages/design-system/theming/src/utils/TokensAccessor/DarkModeTheme.ts [hunk 1]
```diff
@@ -140,7 +140,7 @@ export class DarkModeTheme implements ColorModeTheme {
   private get fgAccent() {
     const color = this.seedColor.clone();
 
-    if (this.seedColor.contrastAPCA(this.bg) <= 60) {
+    if (this.bg.contrastAPCA(this.seedColor) >= -60) {
```
Comment (KelvinOm) [inline app/client/packages/design-system/theming/src/utils/TokensAccessor/DarkModeTheme.ts:143]: It worked correctly. Just wondering why you changed it.

File: app/client/packages/design-system/theming/src/utils/TokensAccessor/LightModeTheme.ts [hunk 2]
```diff
@@ -251,29 +251,24 @@ export class LightModeTheme implements ColorModeTheme {
...
+      tint.oklch.c = 0;
+      shade.oklch.c = 0;
     }
 
-    if (this.seedIsAchromatic) {
-      color.oklch.l = 0.15;
-      color.oklch.c = 0;
-      return color;
+    tint.oklch.l = 0.96;
+    shade.oklch.l = 0.23;
```
Comment (KelvinOm) [inline app/client/packages/design-system/theming/src/utils/TokensAccessor/LightModeTheme.ts:263]: I'm sad because it's not 0.27 like in DarkMode.

**Review by KelvinOm — APPROVED** [review]

File: app/client/packages/design-system/theming/src/utils/TokensAccessor/DarkModeTheme.ts (hunk 1 above)
Comment (ichik) [reply app/client/packages/design-system/theming/src/utils/TokensAccessor/DarkModeTheme.ts:143]: It didn't work correctly. We were getting results that seemed like they are correct by checking the incorrect contrast and then basing our algo on inverting it.
